api/types/swarm/network.go

```diff
@@ -84,6 +84,8 @@ type NetworkSpec struct {
 	Attachable          bool         `json:",omitempty"`
 	Ingress             bool         `json:",omitempty"`
 	IPAMOptions         *IPAMOptions `json:",omitempty"`
+	ConfigFrom          string       `json:",omitempty"`
```

Comment:
Not really a huge fan of this pattern and as is isn't really extendable, but I do see the need for it.

Instead and this be a struct?

```
type NetworkConfigReference struct {
    Network *NetworkReference
}

type NetworkReference {
    Network string
}
```

And then maybe we can change the field name to `NetworkConfigReference` which points to a `*NetworkConfigReference`.
Also maybe a struct for `NetworkReference` here is overkill and that could just be a string on `NetworkConfigReference`.

*note*: not really tied to these names, just that `ConfigFrom` is ok as a flag name, but not very descriptive for an API.